### PR TITLE
[CONTP-402] Store config values in config webhook struct

### PR DESCRIPTION
**Pull request by gabedos**
### What does this PR do?

Follow up to [last PR](https://github.com/DataDog/datadog-agent/pull/29881), by storing relevant config values in struct.

### Motivation

Currently, the config mutator is storing a pointer to the datadogConfig and fetching values as needed. Unfortunately, caching does not work with the config component so we are storing the relevant values in the webhook struct instead.

### Describe how to test/QA your changes

Unit tests should be fine.

### CODE REVIEW

File: pkg/clusteragent/admission/mutate/config/config.go [hunk 1]
```diff
@@ -65,29 +65,37 @@ const (
...
+	resources         []string
+	operations        []admissionregistrationv1.OperationType
+	mode              string
+	wmeta             workloadmeta.Component
+	injectionFilter   mutatecommon.InjectionFilter
+	localServiceName  string
+	traceAgentSocket  string
+	dogStatsDSocket   string
+	socketPath        string
+	typeSocketVolumes bool
```
Comment (clamoriniere) [inline pkg/clusteragent/admission/mutate/config/config.go:80]: maybe for readability, you can split the list of field with an empty line before, or create a sub structure `type config struct`

**Comment by davidor** [comment]
@gabedos the ci/circleci:unit_tests job is failing but it's unrelated to your changes. It should pass if you rebase on top of main.

File: pkg/clusteragent/admission/mutate/config/config.go [hunk 2]
```diff
@@ -65,29 +65,37 @@ const (
...
+		resources:         []string{"pods"},
+		operations:        []admissionregistrationv1.OperationType{admissionregistrationv1.Create},
+		mode:              datadogConfig.GetString("admission_controller.inject_config.mode"),
+		wmeta:             wmeta,
+		injectionFilter:   injectionFilter,
+		localServiceName:  datadogConfig.GetString("admission_controller.inject_config.local_service_name"),
+		traceAgentSocket:  datadogConfig.GetString("admission_controller.inject_config.trace_agent_socket"),
+		dogStatsDSocket:   datadogConfig.GetString("admission_controller.inject_config.dogstatsd_socket"),
+		socketPath:        datadogConfig.GetString("admission_controller.inject_config.socket_path"),
+		typeSocketVolumes: datadogConfig.GetBool("admission_controller.inject_config.type_socket_volumes"),
```
Comment (clamoriniere) [inline pkg/clusteragent/admission/mutate/config/config.go:98]: regrouping the fields coming from the `datadogConfig` in a sub structure can help for the readability

File: pkg/clusteragent/admission/mutate/config/config.go (hunk 1 above)
Comment (gabedos) [reply pkg/clusteragent/admission/mutate/config/config.go:80]: Thank you for the comment. I totally agree this struct was getting a little out of hand.

**Review by clamoriniere — APPROVED** [review]
Nice work 👏

**Pull request merged**